pkg/accumulator/resaccumulator.go

```diff
@@ -94,7 +94,21 @@ func (ra *ResAccumulator) MergeAccumulator(other *ResAccumulator) (err error) {
 	if err != nil {
 		return err
 	}
-	return ra.varSet.MergeSet(other.varSet)
+	err = ra.varSet.MergeSet(other.varSet)
```

Comment:
Just back from dinner. Note to self, because map iteration is non-deterministic and this stops iteration at the first error, this is not always going to work. This method needs to always complete iteration, returning all errors for evaluation by the new clause below.

Reply:
can you comment on why the check on var values, and deciding not to return an error if they match, cannot be done in the body of Var.Merge?

Reply:
i'm not looking at this too deeply, but i thought that was where the fix was going to go

Reply:
> can you comment on why the check on var values, and deciding not to return an error if they match, cannot be done in the body of Var.Merge?

Yes, I would be glad to comment. I have two responses.

1. I am certain it is possible to perform the check entirely within `Var.Merge`. Perhaps the `Var` type could gain a `Value` method that takes a `resaccumulator` as an argument (correspondingly the `resaccumulator` would lose `findVarValueFromResources`). With this approach, `Var.Merge` would need access to the appropriate `resaccumulator`s so it could pass them down to the `Value` method.

2. Based on my read of the abstractions at play (such as I understand them after reading the code for less than a day) the Var type is, today, primarily concerned with recording where a var may be found and that no duplicates exist. The nearest place in the hierarchy of the API (that I could see clearly) that was concerned with retrieving values was the resaccumulator.

Given those two responses, could you comment on which approach hews most closely to your ideals for the codebase @monopole? I'm quite happy to implement this in either fashion (or an altogether different one, such that you are willing to take the time to explain it or otherwise prompt me to discover it on my own).

Reply:
thanks, sounds good.  this error is in the view of the accumulator.